src/eddymotion/viz.py

```diff
@@ -277,3 +278,136 @@ def plot_gradients(
         plt.suptitle(title)
 
     return ax
+
+
+def plot_carpet(
+    nii,
+    gtab,
+    brain_mask=None,
+    sort_by_bval=False,
+    output_file=None,
+    segment_labels=None,
+    detrend=False,
+):
+    """
+    Return carpet plot using niworkflows carpet_plot
+
+    Parameters
+    ----------
+    nii : Nifti1Image
+        DW imaging data
+    gtab : :obj:`GradientTable`
+        DW imaging data gradient data
+    brain_mask : Nifti1Image
+        Boolean or segmentation mask of DW imaging data
+    sort_by_bval : :obj:`bool`
+        Flag to reorder time points by bvalue
+    output_file : :obj:`string`
+        Path to save the plot
+    segment_labels : :obj:`dict`
+        Dictionary of segment labels
+        e.g. {'Cerebral_White_Matter': [2, 41],
+              'Cerebral_Cortex': [3, 42],
+              'Ventricle': [4, 14, 15, 43, 72]}
+    detrend : :obj:`bool`
+        niworkflows plot_carpet detrend flag
+
+    Returns
+    ---------
+    matplotlib GridSpec object
+    """
+    segments = None
+
+    nii_data = nii.get_fdata()
+
+    b0_data = nii_data[..., gtab.b0s_mask]
+    dw_data = nii_data[..., ~gtab.b0s_mask]
+
+    bzero = np.mean(b0_data, -1)
+
+    nii_data_div_b0 = dw_data / (np.expand_dims(bzero, 3).repeat(dw_data.shape[3], 3))
```

Comment:
I believe numpy just needs you to create an axis so that it knows how to cast the array:
```suggestion
    nii_data_div_b0 = dw_data / bzero[..., np.newaxis]
```